Bottom: bddb79278d540c7ef6f2e58f7558f79aa2ca70bc
Top:    5e6977f60ba30b1a75a8c0ee325446ede9c05019
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 15:42:40 +0100

Workaround crash where contour is passed as NULL to poly_PreContour












---

diff --git a/src/polygon.c b/src/polygon.c
index 349886a..9bf6e83 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -237,6 +237,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
